.github/workflows/test.yml

```diff
@@ -0,0 +1,22 @@
+name: Test
+run-name: Run tests
+on: 
+  pull_request:
+    branches:
+      - main
+jobs:
+  run-tests:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Check out repository
+          uses: actions/checkout@v4
+        - name: Install dependencies
+          run: |
+            sudo apt-get update
+            DEBIAN_FRONTEND=noninteractive sudo apt-get install -y autoconf autotools-dev bison catch2 flex g++ libboost-all-dev libicu-dev libtool make pkg-config
```

Comment:
This is a nit, but I'd eventually want this list of packages to be in a shell script, like `scripts/ubuntu_dependencies.sh` or something, assuming there's room for building out other distros/platforms and they all have variations in these package names.